.github/workflows/pathogen-repo-build.yaml.in

```diff
@@ -367,43 +367,56 @@ jobs:
         env:
           AWS_BATCH_JOB_ID: ${{ needs.run-build.outputs.AWS_BATCH_JOB_ID }}
         run: |
-          # When a running process is to be cancelled (e.g. due to either user
-          # request or job timeout), GitHub Actions sends it SIGINT and waits
-          # 7.5s for it to exit.¹  We immediately detach on interrupt, letting
-          # the next wait-N job in our chain pick it back up (if this job timed
-          # out) or the cancellation job (if by user request).
-          #
-          # ¹ <https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow#steps-github-takes-to-cancel-a-workflow-run>
-          exec nextstrain build \
+          # Using timeout to detach from build before we hit the job.timeout-minutes
+          # to avoid cancelling the job in order to avoid a "Cancelled" status
+          # for the whole workflow.
+          # <https://github.com/nextstrain/.github/issues/129#issuecomment-2956350306>
+          exit_status=0
+          timeout 359m nextstrain build \
             --aws-batch \
             --attach "$AWS_BATCH_JOB_ID" \
             --detach-on-interrupt \
-            --no-download
+            --no-download || exit_status=$?
+
+          if [[ $exit_status == 0 ]]; then
+            conclusion="success"
+          elif [[ $exit_status == 124 ]]; then
+            # 124 is the exit code for timed out commands
+            # Mask the exit code so that the job concludes successfully
+            # and we can continue the next wait-N job.
+            exit_status=0
+            conclusion="timeout"
+          else
+            conclusion="failure"
+          fi
+
+          echo "conclusion=$conclusion" | tee -a "$GITHUB_OUTPUT"
+          exit $exit_status
 
     # Emit a "conclusion" output for the job that's based on the built-in
     # conclusion (success, failure, cancelled) of the "attach" step above.
     # This is the conclusion we care about for the job.
```

Comment:
This comment is no longer correct since the built-in step conclusion is not being used any more. It'd be good to update it.